src/psyclone/psyir/transformations/loop_trans.py

```diff
@@ -112,7 +112,17 @@ def validate(self, node, options=None):
                 if isinstance(item, self.excluded_node_types):
                     raise TransformationError(
                         f"Nodes of type '{type(item).__name__}' cannot be "
-                        f"enclosed by a {self.name} transformation")
+                        f"enclosed by a {self.name} transformation (use "
+                        f"the 'node-type-check: False' option to accept them "
+                        f"at your own risk)")
+
+            for assignment in node.walk(Assignment):
+                if assignment.is_pointer:
+                    raise TransformationError(
+                        f"'{type(self).__name__}' can not be applied to nodes"
+                        f" that contain pointer assignments by default (use "
```

Comment:
Include `assignment.debug_string()` in the output so the user can more easily identify the location?

Reply:
Since I want to use the messages as comments in code, and then the debug string is unecessary. I went with the same pattern than ArrayAssignment2Loop with the "verbose" option adding the preceding comment and the error having a LazyString with the debug_string (so its not generate if not requested by the except).

Reply:
Also great.

Reply:
Nice.